docs/reference/contrib/settings.md

```diff
@@ -0,0 +1,379 @@
+
+Settings are editable by administrators within the Wagtail admin, and can be
+accessed in code as well as in templates.
+
+## Installation
+
+Add `wagtail.contrib.settings` to your `INSTALLED_APPS`:
+
+```python
+INSTALLED_APPS += [
+    'wagtail.contrib.settings',
+]
+```
+
+**Note:** If you are using `settings` within templates, you will also need to
+update your `TEMPLATES` settings (discussed later in this page).
+
+## Defining settings
+
+Create a model that inherits from either:
+
+* `BaseGenericSetting` (for generic settings across all sites)
+* `BaseSiteSetting` (for site-specific settings)
+
+and register it using the `register_setting` decorator:
+
+```python
+from django.db import models
+from wagtail.contrib.settings.models import (
+    BaseGenericSetting,
+    BaseSiteSetting,
+    register_setting,
+)
+
+@register_setting
+class GenericSocialMediaSettings(BaseGenericSetting):
+    facebook = models.URLField()
+
+@register_setting
+class SiteSpecificSocialMediaSettings(BaseSiteSetting):
+    facebook = models.URLField()
+```
+
+Links to your settings will appear in the Wagtail admin 'Settings' menu.
+
+## Edit handlers
+
+Settings use edit handlers much like the rest of Wagtail.
+Add a `panels` setting to your model defining all the edit handlers required:
+
+```python
+@register_setting
+class GenericImportantPages(BaseGenericSetting):
+    donate_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+    sign_up_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+
+    panels = [
+        FieldPanel('donate_page'),
+        FieldPanel('sign_up_page'),
+    ]
+
+@register_setting
+class SiteSpecificImportantPages(BaseSiteSetting):
+    donate_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+    sign_up_page = models.ForeignKey(
+        'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+'
+    )
+
+    panels = [
+        FieldPanel('donate_page'),
+        FieldPanel('sign_up_page'),
+    ]
+```
+
+You can also customize the editor handlers [like you would do for Page model](/advanced_topics/customisation/page_editing_interface.html#customising-the-tabbed-interface) with a custom `edit_handler` attribute:
+
+```python
+from wagtail.admin.panels import TabbedInterface, ObjectList
+
+@register_setting
+class MySettings(BaseGenericSetting):
+    # ...
+    first_tab_panels = [
+        FieldPanel('field_1'),
+    ]
+    second_tab_panels = [
+        FieldPanel('field_2'),
+    ]
+
+    edit_handler = TabbedInterface([
+        ObjectList(first_tab_panels, heading='First tab'),
+        ObjectList(second_tab_panels, heading='Second tab'),
+    ])
+```
+
+## Appearance
+
+You can change the label used in the menu by changing the
+`django.db.models.Options.verbose_name` of your model.
```

Comment:
Spelling out the full path to ``verbose_name`` feels a little verbose to me!

```suggestion
`verbose_name` of your model.
```

Reply:
Agreed - a lot of the copy on this page is lifted straight from the current `.rst` version with minimal amends.